sklearn/utils/tests/test_validation.py

```diff
@@ -346,6 +346,7 @@ def test_check_array_dtype_stability():
     assert_equal(check_array(X, ensure_2d=False).dtype.kind, "i")
 
 
+@pytest.mark.skip(reason="warn_on_dtype is deprecated from 'check_array'")
```

Comment:
Don't skip the tests that use `warn_on_dtype`: there are still valid checks inside.
Instead, please ignore the deprecation warning with something like:

`@pytest.mark.filterwarnings("ignore: 'warn_on_dtype' is deprecated from")  # 0.23`